adapters/httpapi_compact.c

```diff
@@ -1060,6 +1093,11 @@ static HTTPAPI_RESULT ReadHTTPResponseBodyFromXIO(HTTP_HANDLE_DATA* http_instanc
                 {
                     /*Codes_SRS_HTTPAPI_COMPACT_21_033: [ If the whole process succeed, the HTTPAPI_ExecuteRequest shall retur HTTPAPI_OK. ]*/
                     result = HTTPAPI_OK;
+
+                    if (on_chunk_received != NULL)
```

Comment:
if on_chunk_received is not null, the http response will be delivered in responseContent as well as via on_chunk_received callback. Do we really have use cases that both are needed?

Reply:
Yes, this is indeed the case. 
A typical case is, you need to write the chunked the data to Push/PullAudioOuputStream, and meanwhile return the final data as SpeechSynthesisResult

Reply:
For this http api, it looks weird to have different ways to get the same result. If the same data needs to be fed into different consumers, the caller, e.g. tts_rest_adapter, should be responsible for forwarding the same data, e.g. audio stream or syntheisresult,

Reply:
This pattern is actually very similar with the TTS API by Rob's spec, like below case,
User gets the same data from different ways: events and result

    auto synthesizer = SpeechSynthesizer::FromConfig(config, nullptr);
    synthesizer->Synthesizing += [](const SpeechSynthesisEventArgs& e) {
        auto audioData = e.Result->GetAudioData();
        DoSomethingWithAudioInVector(audioData);
    };

    auto result = synthesizer->SpeakTextAsync("{{{text1}}}").get();

Reply:
That is API level. Creating duplicated audio data twice from the lowest network layer is not memory friendly. The copy should be delayed as late as possible.

Reply:
Indeed, I do not think there is needs to add the on_chunk_received callback here, because the result is directly returned in buffer anyway (so no performance improvement). The callback only makes threading more complicated. Since the callback runs in the same thread, the callback behavior might affect the run time of this method. If the upper layer caller needs to use the data for different purpose, the caller should multiplex the data there, but not at this level.

Reply:
Updated per our discussion. We can call if you still have question

Reply:
Thanks!